src/dmd/link.d

```diff
@@ -569,38 +569,21 @@ public int runLINK()
             argv.push("-Xlinker");
             argv.push("--gc-sections");
         }
-        /* Add each library, prefixing it with "-l".
+        /* Add each library, prefixing it with "-l" if necessary.
          * The order of libraries passed is:
-         *  1. any libraries passed with -L command line switch
-         *  2. libraries specified on the command line
+         *  1. static libraries ending with *.a
+         *  2. link switches passed with -L command line switch
          *  3. libraries specified by pragma(lib), which were appended
          *     to global.params.libfiles.
-         *  4. link switches, that may also contain -l libraries
-         *  5. standard libraries.
+         *  4. standard libraries.
```

Comment:
How does this PR affect dynamic libraries passed directly on the command line without a flag?

Reply:
Not affecting at all.
Arguments not starting with `-L` and ending with `.so` are in `global.params.dllfiles`.
They are passed to the linker right before std lib link flags as before this PR and before the PR that introduced this regression.

Reply:
Sorry that's not right. I double checked. see next comment with argument order evolution